pkgs/development/compilers/crystal/default.nix

```diff
@@ -38,8 +38,8 @@ let
     '';
   };
 
-  generic = { version, sha256, binary, doCheck ? true }:
-  let compiler = stdenv.mkDerivation rec {
+  generic = ({ version, sha256, binary, doCheck ? true, extraBuildInputs ? [] }:
+  lib.fix (compiler: stdenv.mkDerivation {
```

Comment:
Do you have a reference to documentation for lib.fix?

Reply:
it's documented at https://github.com/NixOS/nixpkgs/blob/master/lib/fixed-points.nix#L3-L25